numpyro/infer/util.py

```diff
@@ -209,10 +209,7 @@ def _unconstrain_reparam(params, site):
             if jnp.ndim(p) > expected_unconstrained_dim:
                 p = p[i]
 
-        if support is constraints.real or (
-            isinstance(support, constraints.independent)
-            and support.base_constraint is constraints.real
-        ):
```

Comment:
could you revert this change? the current logic also supports `real_matrix` etc.